### ISSUE TITLE
It takes a long time to train semantic segmentation models

### ISSUE THREAD

**Issue by moniec1717** [yanx27/Pointnet_Pointnet2_pytorch#278]
![Screenshot 2024-10-14 161459](https://github.com/user-attachments/assets/71395e55-c92d-4a4e-9332-5a8b3e2cd2cd)
when I train my own dataset (point clouds of bridges and roads), it takes 15h for just one epoch. 
How can I improve my model? Which parameters should I change? 
I have already tried changing the parameter stride and block_size, but it didn't work on my dataset.
Can anyone help me out, thanks a lot!

**Comment by kvta1** [comment]
I am having similar issues, the code seems to not want to run on my GPU. Have you already found a fix?

**Comment by kvta1** [comment]
Are you training on GPU? You can increase batch_size

**Comment by moniec1717** [comment]
> Are you training on GPU? You can increase batch_size

I've adjusted the batch size, but no great improvement. I thought maybe the GPU wasn't being used, but after checking the code and seeing that my computer’s memory did use it, I'm not really sure where else things could be improved.

**Comment by jianuo1128** [comment]
@moniec1717 have you ever solved this issue？

**Comment by moniec1717** [comment]
> [@moniec1717](https://github.com/moniec1717) have you ever solved this issue？

No, I haven't yet identified the issue with the models. As my main method for my paper is based on KPConv, I decided not to continue training the datasets using Pointnet.

**Comment by Youyuan287** [comment]
The code for PointNet and PointNet++ is indeed quite outdated. Consider trying newer models such as X-3D or DeLA. These newer models may offer advantages in performance and efficiency.
